receiver/vcenterreceiver/testdata/metrics/expected.yaml

```diff
@@ -2636,6 +2636,12 @@ resourceMetrics:
         - key: vcenter.vm.name
           value:
             stringValue: CentOS 7
+        - key: vcenter.resource_pool.name
+          value:
+            stringValue: Resources
+        - key: vcenter.resource_pool.inventory_path
+          value:
+            stringValue: /Datacenter/host/Cluster/Resources
```

Comment:
Is there a bug fix or other change here, in addition to the optimization? If not, why would the expected result change?

Reply:
@djaglowski I think I would argue that I improved the tests here by having better XML responses being returned (which resulted in updated expected test results). Now the tests show the new resource pool attributes being added to VMs.

Reply:
Thanks for explaining. Would it be possible to submit the changes to the mock server separately? I'm sure they're good changes but it would be helpful to validate the refactoring independently.

Reply:
@djaglowski I can do this. 

I do want to call out that the reason that these changes are there are that I think they pertain to the testing of this block of work (the VM code). 

I made changes, the tests were broken.
So I started to update the tests, with new recordings from my new requests. But then I noticed that the expected results didn't seem to match what the code was doing (which was what I WAS expecting....if that makes any sense 😄 ).

Like I said, I can work around this and modify this PR to remove the expected results modifications. But it will make the tests in a less than optimal state IMO.

Reply:
Will the test changes work without the other changes in this PR? I was thinking that we would first update tests, then rebase this PR so it is only refactoring for performance.

Reply:
@djaglowski let me give that a try. I think that should work and not be too difficult to implement. 👍

Reply:
@djaglowski rebased off the main which contained the merged in fixes from the previous PR